error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/EagerStringFormattingTest.java

```diff
@@ -0,0 +1,286 @@
+package tech.picnic.errorprone.bugpatterns;
+
+import com.google.errorprone.BugCheckerRefactoringTestHelper;
+import com.google.errorprone.BugCheckerRefactoringTestHelper.TestMode;
+import com.google.errorprone.CompilationTestHelper;
+import org.junit.jupiter.api.Test;
+
+final class EagerStringFormattingTest {
```

Comment:
Shouldn't we also check for the case when the potential `String#format` invocations is stored and passed in a variable to the "parent" method?
```patch
Index: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/EagerStringFormattingTest.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>ISO-8859-1
===================================================================
diff --git a/error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/EagerStringFormattingTest.java b/error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/EagerStringFormattingTest.java
--- a/error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/EagerStringFormattingTest.java	(revision caff523200614c14f23100fe4957dbf3d020bb4a)
+++ b/error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/EagerStringFormattingTest.java	(date 1735330311287)
@@ -65,6 +65,9 @@
             "",
             "    requireNonNull(\"never-null\");",
             "    requireNonNull(\"never-null\", () -> String.format(\"Format string: %s\", nonFinalField));",
+            "    // BUG: Diagnostic matches: DEFER_PARAM",
+            "    String loggedMessage = String.format(\"Never-null format string: %s\", nonFinalField)",
+            "    requireNonNull(loggedMessage);",
             "    // BUG: Diagnostic matches: VACUOUS",
             "    requireNonNull(String.format(\"Never-null format string: %s\", nonFinalField));",
             "    // BUG: Diagnostic matches: VACUOUS",

```

Reply:
>  is stored and passed in a variable to the "parent" method?

Or even go far and check if it's returned from another method invocation 👀

Reply:
Indeed, none of these cases are currently covered. For now I'll call out these limitations in a comment; the check/PR is already complex enough as it is.